beets/dbcore/types.py

```diff
@@ -190,6 +190,38 @@ class String(Type):
     query = query.SubstringQuery
 
 
+class StringList(Type):
+    """A List of Unicode strings type, stored in a string
+    """
+    sql = u'TEXT'
+    query = query.SubstringQuery
+
+    def from_sql(self, sql_value):
+        return self.parse(sql_value)
+
+    def to_sql(self, value):
+        return self.format(value)
+
+    def format(self, value):
+        """Given a value of this type, produce a Unicode string
+        representing the value. This is used in template evaluation.
+        """
+
+        if isinstance(value, list):
```

Comment:
Any particular reason why this check is necessary? If the value is somehow taking on a string value, we should probably stop that! It will cause other problems—for example, `to_sql` is likely to do something unpredictable. As with the other types, we should enforce that the underlying value must always be a tuple of strings.